src/recordlinker/linking/mpi_service.py

```diff
@@ -109,3 +111,40 @@ def insert_blocking_keys(
     if commit:
         session.commit()
     return values
+
+
+def get_patient_by_reference_id(
+    session: orm.Session, reference_id: uuid.UUID
+) -> models.Patient | None:
+    """
+    Retrieve the Patient by their reference id
+    """
+    query = select(models.Patient).where(models.Patient.reference_id == reference_id)
+    return session.scalar(query)
```

Comment:
Is this the same as `session.execute(query).scalars()`?

Reply:
Its not, [session.scalar(query)](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.scalar) is a shortcut to execute a query and return the single result as an object, or None if nothing was found.  Its expecting at most 1 row in the result set.

[session.execute(query).scalars()] is an operation to perform on a result set that transforms the sequence of [Rows](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Row) to a sequence of [ScalarResults](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.ScalarResult).